### DISCUSSION TITLE
Is any good way to make deb file

### DISCUSSION THREAD

**Discussion by abubaca4** [transmission/transmission#4745] [Q&A] [answered]
Now I use 3.0 from debian 11 repo
As I know newer version of software will be only in debian 12, debian make only secure updates in stable systems
I want update transmission-daemon to 4.0
I don't want use docker image due performance issues on my single board
I want use package manager, not make install
What should be config of cmake and CheckInstall to make transmission-daemon package similar to repo package?
It look like at least 4.0 require libdeflate instead of zlib. And I don't know what versions needed

Or I just have to remove repo package and use make install?

**Comment by ckerr** [comment]
There's some discussion about this going on at https://github.com/transmission/transmission/discussions/4742

**Reply by abubaca4** [reply]
I look it before create this one and don't found dep list(and that contains a lot of basic question about depedens)
Can you give dep list for 4.0.0 daemon with minimum versions?

**Comment by fredo-47** [comment]
If libdeflate is not found, transmission brings its own.

**Reply by abubaca4** [reply]
I noticed that builded binary much bigger than current from repo. So we need only openssl, other libs included in binary?

**Comment by abubaca4** [comment, marked as answer]
As a result for build in debian 11 I need `cmake make g++`
I finded dependency packages `libcurl4-openssl-dev libssl-dev libb64-dev libfmt-dev libevent-dev libnatpmp-dev libpsl-dev libminiupnpc-dev libutfcpp-dev libsystemd-dev`
Most of them can be skipped and last versions will be included in transmission binary. But if you skip `libsystemd-dev` you can't use Type=notify in systemd daemon
deb package can be created tools like checkinstall or debhelper but in the end I just put result file in /usr/local/bin and change path of binary in systemd daemon
Required packages versions can be find in [cmake file](https://github.com/transmission/transmission/blob/main/CMakeLists.txt#L32)
